packages/theme-check-common/src/checks/valid-render-snippet-params/index.ts

```diff
@@ -0,0 +1,135 @@
+import { LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+import { LiquidNamedArgument, RenderMarkup } from '@shopify/liquid-html-parser';
+import { toLiquidHtmlAST } from '@shopify/liquid-html-parser';
+import { getSnippetDefinition, LiquidDocParameter } from '../../liquid-doc/liquidDoc';
+import { isLiquidString } from '../utils';
+import { getDefaultValueForType } from '../../liquid-doc/utils';
+
+export const ValidRenderSnippetParams: LiquidCheckDefinition = {
+  meta: {
+    code: 'ValidRenderSnippetParams',
+    name: 'Valid Render Snippet Parameters',
+
+    docs: {
+      description:
+        'This check ensures that all required parameters are provided when rendering a snippet and that no unknown parameters are used.',
+      recommended: true,
+      url: 'https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/valid-render-snippet-params',
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.WARNING,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    function partitionParameters(
+      liquidDocParameters: Map<string, LiquidDocParameter>,
+      providedParams: LiquidNamedArgument[],
+    ) {
+      const providedParameters = new Map(providedParams.map((arg) => [arg.name, arg]));
+
+      const missingRequiredParams: LiquidDocParameter[] = [];
+      const unknownProvidedParams: LiquidNamedArgument[] = [];
+
+      // Check required params
+      for (const param of liquidDocParameters.values()) {
+        if (param.required && !providedParameters.has(param.name)) {
+          missingRequiredParams.push(param);
+        }
+      }
+
+      // Check provided params
+      for (const arg of providedParameters.values()) {
+        const liquidDocParamDef = liquidDocParameters.has(arg.name);
+        if (!liquidDocParamDef) {
+          unknownProvidedParams.push(arg);
+        }
```

Comment:
In the PR above this we expand this condition to perform type validations if the `paramDef` exists